### PR TITLE
/fundraiser Controller

### PR DESCRIPTION
**Pull request by rraihansaputra**
POST /fundraiser

GET /fundraiser/<fundraiser_id>
PUT /fundraiser/<fundraiser_id>
DELETE /fundraiser/<fundraiser_id>

### CODE REVIEW

**Comment by thaumasiotes** [comment]
When I run the migrations, I get a database with timestamps that lack a time zone:

```
                                         Table "public.fundraisers"
   Column    |            Type             | Collation | Nullable |                 Default                 
-------------+-----------------------------+-----------+----------+-----------------------------------------
 id          | integer                     |           | not null | nextval('fundraisers_id_seq'::regclass)
 title       | character varying(255)      |           | not null | 
 description | text                        |           | not null | 
 amount      | integer                     |           | not null | 
 deadline    | timestamp without time zone |           | not null | 
 created_at  | timestamp without time zone |           | not null | 
 creator_id  | integer                     |           | not null |
```
Some of these, like `deadline`, seem like they need the accuracy of a marked timezone.

**Comment by rraihansaputra** [comment]
> When I run the migrations, I get a database with timestamps that lack a time zone:
> 
> ```
>                                          Table "public.fundraisers"
>    Column    |            Type             | Collation | Nullable |                 Default                 
> -------------+-----------------------------+-----------+----------+-----------------------------------------
>  id          | integer                     |           | not null | nextval('fundraisers_id_seq'::regclass)
>  title       | character varying(255)      |           | not null | 
>  description | text                        |           | not null | 
>  amount      | integer                     |           | not null | 
>  deadline    | timestamp without time zone |           | not null | 
>  created_at  | timestamp without time zone |           | not null | 
>  creator_id  | integer                     |           | not null |
> ```
> 
> Some of these, like `deadline`, seem like they need the accuracy of a marked timezone.

I've tried to put in accurate UTC timezone in the code (on `resources/Fundraiser.py`) but on local testing, seems like marshmallow/sqlalchemy still doesn't pass on the timezone designation. This needs to be fixed in the future.

File: server/resources/Fundraiser.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        if user.id != fundraiser.creator_id:
+            return {"message": INVALID_CREDENTIALS}, 401
+
+        fundraiser.delete_from_db()
+        return {"message": FUNDRAISER_DELETED}, 200
+
+
+    @classmethod
+    @jwt_required
+    def put(cls, fundraiser_id: int):
```
Comment (rraihansaputra) [inline server/resources/Fundraiser.py:113]: @thaumasiotes @aeciorc I'd like your comment on the implementation of the PUT method. It feels repetitive to me. Looking for suggestions from both of you.

**Comment by rraihansaputra** [comment]
If this all looks correct besides the timezone issue, I would be happy to merge it as @thaumasiotes's work on issue #20 needs to be rebased from work on this PR.

File: server/resources/Fundraiser.py (hunk 1 above)
Comment (thaumasiotes) [reply server/resources/Fundraiser.py:113]: I see your point, but I'm not sure what to do about it...

**Comment by thaumasiotes** [comment]
The postgres documentation [suggests](https://www.postgresql.org/docs/10/datatype-datetime.html#DATATYPE-TIMEZONES) that `timestamp without time zone` really means "UTC timestamp" internally, which would be fine.

The rest of the commit seems fine to me, but I'm also interested in @aeciorc's review.

**Comment by rraihansaputra** [comment]
Looks like I've found out how to solve the timezone problem. SQLAlchemy can define the column to store the timezone or not. Any conversion will be handled directly by PostgreSQL depending on the database timezone. I've created the migrations on the [the latest commit (6d211dc)](https://github.com/hatchways/team-broccoli/pull/30/commits/6d211dc08f081bd814ae54cc67d1d70026bab0d3). I'm not sure whether to roll this into this PR or to create a separate PR for this issue.

File: server/resources/Fundraiser.py (hunk 1 above)
Comment (aeciorc) [reply server/resources/Fundraiser.py:113]: Seems fine to me, other than combining a couple statements into one-liners, there's not much you can do. On a separate note, are those dict returns converted to json?

Comment (rraihansaputra) [reply server/resources/Fundraiser.py:113]: OK, will leave it as it is. Thanks for the review!

Yep, it's converted to JSON. `flask-restful` is handling the conversion from dict to JSON.

**Pull request merged**